src/Internal/Payment/State/InitialState.php

```diff
@@ -46,69 +42,45 @@ class InitialState extends AbstractPaymentState {
 
-	/**
-	 * Payment request service.
-	 *
-	 * @var PaymentRequestService
-	 */
-	private $payment_request_service;
-
 	/**
 	 * Class constructor, only meant for storing dependencies.
 	 *
 	 * @param StateFactory                 $state_factory           Factory for payment states.
 	 * @param OrderService                 $order_service           Service for order-related actions.
 	 * @param WC_Payments_Customer_Service $customer_service        Service for managing remote customers.
 	 * @param Level3Service                $level3_service          Service for Level3 Data.
-	 * @param PaymentRequestService        $payment_request_service Connection with the server.
 	 */
 	public function __construct(
 		StateFactory $state_factory,
 		OrderService $order_service,
 		WC_Payments_Customer_Service $customer_service,
-		Level3Service $level3_service,
-		PaymentRequestService $payment_request_service
+		Level3Service $level3_service
 	) {
 		parent::__construct( $state_factory );
 
-		$this->order_service           = $order_service;
-		$this->customer_service        = $customer_service;
-		$this->level3_service          = $level3_service;
-		$this->payment_request_service = $payment_request_service;
+		$this->order_service    = $order_service;
+		$this->customer_service = $customer_service;
+		$this->level3_service   = $level3_service;
 	}
 
 	/**
 	 * Initialtes the payment process.
 	 *
 	 * @param PaymentRequest $request    The incoming payment processing request.
-	 * @return CompletedState            The next state.
+	 * @return AbstractPaymentState      The next state.
 	 * @throws StateTransitionException  In case the completed state could not be initialized.
 	 * @throws ContainerException        When the dependency container cannot instantiate the state.
 	 * @throws Order_Not_Found_Exception Order could not be found.
 	 * @throws PaymentRequestException   When data is not available or invalid.
 	 */
-	public function process( PaymentRequest $request ) {
-		$context  = $this->get_context();
-		$order_id = $context->get_order_id();
-
+	public function start_processing( PaymentRequest $request ) {
 		// Populate basic details from the request.
 		$this->populate_context_from_request( $request );
 
 		// Populate further details from the order.
 		$this->populate_context_from_order();
 
-		// Payments are currently based on intents, request one from the API.
-		try {
-			$intent = $this->payment_request_service->create_intent( $context );
-		} catch ( Invalid_Request_Parameter_Exception | Extend_Request_Exception | Immutable_Parameter_Exception $e ) {
-			return $this->create_state( SystemErrorState::class );
-		}
-
-		// Intent available, complete processing.
-		$this->order_service->update_order_from_successful_intent( $order_id, $intent, $context );
-
-		// If everything went well, transition to the completed state.
-		return $this->create_state( CompletedState::class );
+		return $this->create_state( VerifyState::class );
```

Comment:
In this case we're not done with processing, so this should be:
```suggestion
		$next_state = $this->create_state( VerifyState::class );
		return $next_state->process();
```

Reply:
Addressed here: https://github.com/Automattic/woocommerce-payments/pull/7471/commits/fa1721e9d3fbd25e4c5b63fb88d5b21bf6d8e1f6